### PR TITLE
better script error handling.

### PR DESCRIPTION
**Pull request by farfromrefug**
Also pass the error in msg so that it can be handled in flows.handleError

### CODE REVIEW

**Comment by farfromrefug** [comment]
example script:
```javascript
const tf = global.get('tf');
const model = tf.sequential();
model.add(tf.layers.dense({units: 1, inputShape: [1]}));

// Prepare the model for training: Specify the loss and the optimizer.
model.compile({loss: 'meanSquaredError', optimizer: 'sgd'});

// Generate some synthetic data for training.
const xs = tf.tensor2d([1, 2, 3, 4], [4, 1]);
const ys = tf.tensor2d([1, 3, 5, 7], [4, 1]);

// Train the model using the data.
model.fit(xs, ys).then(() => {
// Use the model to do inference on a data point the model hasn't seen before:
    node.send({
        payload:model.predict(tf.tensor2d([5], [1, 1])).flatten()
    })
});
```

output before:
```
/Volumes/data/dev/aryballe/aryballe-studio/app/node_modules/@tensorflow/tfjs-core/dist/ops/array_ops.js:56
            throw new Error('Error creating a new Tensor2D: values must be number[][] ' +
            ^

Error: Error creating a new Tensor2D: values must be number[][] or flat/TypedArray
    at Object.ArrayOps.tensor2d (/Volumes/data/dev/aryballe/aryballe-studio/app/node_modules/@tensorflow/tfjs-core/dist/ops/array_ops.js:56:19)
    at evalmachine.<anonymous>:10:15
    at evalmachine.<anonymous>:20:3
    at ContextifyScript.Script.runInContext (vm.js:53:29)
    at FunctionNode.<anonymous> (/Volumes/data/dev/aryballe/aryballe-studio/app/node_modules/node-red/nodes/core/core/80-function.js:209:33)
    at emitOne (events.js:115:13)
    at FunctionNode.emit (events.js:210:7)
    at FunctionNode.Node.receive (/Volumes/data/dev/aryballe/aryballe-studio/app/node_modules/node-red/red/runtime/nodes/Node.js:215:14)
    at ButtonNode.Node.send (/Volumes/data/dev/aryballe/aryballe-studio/app/node_modules/node-red/red/runtime/nodes/Node.js:137:22)
    at EventEmitter.handler (/Volumes/data/dev/aryballe/aryballe-studio/app/node_modules/node-red-dashboard/ui.js:267:22)
```

output after : 
```
/Volumes/data/dev/aryballe/aryballe-studio/app/node_modules/@tensorflow/tfjs-core/dist/ops/array_ops.js:56
            throw new Error('Error creating a new Tensor2D: values must be number[][] ' +
            ^

Error: Error creating a new Tensor2D: values must be number[][] or flat/TypedArray
    at Object.ArrayOps.tensor2d (/Volumes/data/dev/aryballe/aryballe-studio/app/node_modules/@tensorflow/tfjs-core/dist/ops/array_ops.js:56:19)
    at tensorflow_da38083d.f04568
    at tensorflow_da38083d.f04568:9:3
    at ContextifyScript.Script.runInContext (vm.js:53:29)
    at FunctionNode.<anonymous> (/Volumes/data/dev/aryballe/aryballe-studio/app/node_modules/node-red/nodes/core/core/80-function.js:209:33)
    at emitOne (events.js:115:13)
    at FunctionNode.emit (events.js:210:7)
    at FunctionNode.Node.receive (/Volumes/data/dev/aryballe/aryballe-studio/app/node_modules/node-red/red/runtime/nodes/Node.js:215:14)
    at ButtonNode.Node.send (/Volumes/data/dev/aryballe/aryballe-studio/app/node_modules/node-red/red/runtime/nodes/Node.js:137:22)
    at EventEmitter.handler (/Volumes/data/dev/aryballe/aryballe-studio/app/node_modules/node-red-dashboard/ui.js:267:22)
```

File: nodes/core/core/80-function.js [hunk 1]
```diff
@@ -204,7 +204,12 @@ module.exports = function(RED) {
         }
         var context = vm.createContext(sandbox);
         try {
-            this.script = vm.createScript(functionText);
+            this.script = vm.createScript(functionText, {
+                filename: this.name + '_' + this.id, // filename for stack traces
```
Comment (knolleary) [inline nodes/core/core/80-function.js]: I think the virtual filename could be a bit more helpful to the user - remember that `this.name` may be blank.

How about `'Function Node:'+this.id+(this.name?' ['+this.name+']':'')`

which would give: 

```
Error: Error creating a new Tensor2D: values must be number[][] or flat/TypedArray
    at Object.ArrayOps.tensor2d (/Volumes/data/dev/aryballe/aryballe-studio/app/node_modules/@tensorflow/tfjs-core/dist/ops/array_ops.js:56:19)
    at Function Node:da38083d.f04568 [tensorflow]
    at Function Node:da38083d.f04568 [tensorflow]:9:3
```

Comment (farfromrefug) [reply nodes/core/core/80-function.js]: @knolleary great idea. Will change that. Was also thinking about removing the "unwanted" part from the stack (from node-red). What do you think?

Comment (knolleary) [reply nodes/core/core/80-function.js]: Yup, if we're adding the stack it should be as useable as possible - doesn't need to be full of internal details

**Comment by farfromrefug** [comment]
@knolleary i fixed it and tested the output:
```
Error : Cannot read property 'Measure' of undefined Function Node:7b04a0cd.df57c [GetMeasures]:10
        return new compute.Measure(r,
                          ^

TypeError: Cannot read property 'Measure' of undefined
    at msg.payload.records.map.r (Function Node:7b04a0cd.df57c [GetMeasures]:10:27)
    at Array.map (native)
    at Function Node:7b04a0cd.df57c [GetMeasures]:9:21
```

File: nodes/core/core/80-function.js (hunk 1 above)
Comment (knolleary) [reply nodes/core/core/80-function.js]: Hi @farfromrefug - can we get the filename field updated here and there's a rogue console.log that needs removing. Thanks!

Comment (farfromrefug) [reply nodes/core/core/80-function.js]: @knolleary should be good now

File: nodes/core/core/80-function.js [hunk 2]
```diff
@@ -204,7 +204,12 @@ module.exports = function(RED) {
         }
         var context = vm.createContext(sandbox);
         try {
-            this.script = vm.createScript(functionText);
+            this.script = vm.createScript(functionText, {
+                filename: 'Function Node:'+this.id+(this.name?' ['+this.name+']':'', // filename for stack traces
```
Comment (knolleary) [inline nodes/core/core/80-function.js]: Missing a `)`

File: nodes/core/core/80-function.js [hunk 3]
```diff
@@ -219,7 +224,14 @@ module.exports = function(RED) {
                         this.status({fill:"yellow",shape:"dot",text:""+converted});
                     }
                 } catch(err) {
+                    //remove unwanted part 
+                    const index = err.stack.search(/\n\s*at ContextifyScript.Script.runInContext/);
```
Comment (knolleary) [inline nodes/core/core/80-function.js]: We still support Node 4 in this release. Please use var rather than const.

**Comment by knolleary** [comment]
Have fixed up the remaining issue manually. Thanks!

**Pull request merged**